src/editor.rs

```diff
@@ -1,24 +1,141 @@
+		self.terminal.draw(|f| root_pane.draw_widgets(f.size(), f))
+	}
+
+	/// Clears the screen
+	pub fn clear(&mut self) -> Result<(), io::Error> {
+		self.terminal.clear()
+	}
+
+	/// Read user input events passed to the Editor as well as update the Lua
+	/// interpreter state.
+	pub fn update(&mut self) -> crossterm::Result<()> {
+		Ok(match read()? {
+			Event::Key(event) => {
+				if event == KeyCode::Char('q').into() {
+					self.running = false;
+				} else if event == KeyCode::Backspace.into() {
+					(*self.root_pane).borrow_mut().delete_backwards_at_cursor(1);
+				} else if event == KeyCode::Delete.into() {
+					(*self.root_pane).borrow_mut().delete_forwards_at_cursor(1);
+				} else {
+					(*self.root_pane)
+						.borrow_mut()
+						.insert_at_cursor(String::from(match event.code {
+							KeyCode::Char('a') => "a",
+							KeyCode::Char('b') => "b",
+							KeyCode::Char('c') => "c",
+							KeyCode::Char('d') => "d",
+							KeyCode::Char('-') => "-",
+							KeyCode::Char(' ') => " ",
+							KeyCode::Char('(') => "(",
+							KeyCode::Char(')') => ")",
+							_ => "hello! ",
+						}));
+				}
+				// self.root_window.window.insert_at_cursor(event.code);
+				self.draw().ok();
+			}
+			Event::Mouse(event) => println!("{:?}", event),
+			Event::Resize(_, _) => {
+				self.terminal
+					.autoresize()
+					.expect("Cannot reload the terminal successfully");
+				self.draw().ok();
+			}
+		})
+	}
+}
+
+impl Drop for Editor {
+	fn drop(&mut self) {
+		execute!(stdout(), LeaveAlternateScreen).unwrap();
+		disable_raw_mode().unwrap();
 	}
 }
 
 impl Default for Editor {
 	fn default() -> Self {
-		Editor {
-			buffers: vec![],
+		enable_raw_mode().unwrap();
+		let stdout = io::stdout();
+		let backend = CrosstermBackend::new(stdout);
+		// TODO: WE NEED TO MOVE BACKENDS SOMEWHERE ELSE
+		let language_lua = unsafe { tree_sitter_lua() };
+
+		let buffers: Vec<Rc<RefCell<Buffer>>> = vec![Rc::new(RefCell::new(Buffer::new(
+			String::from("-- This buffer is for text that is not saved, and for Lua evaluation\n-- Use this to interact with the built-in Lua interpreter.\nfunction hello()\n  print('hello, world!')\nend"),
+			String::from("*scratch*"),
+			Some(language_lua),
+		)))];
+		let buffer = buffers[0].clone();
+
+		let editor = Editor {
+			root_pane: Rc::new(RefCell::new(Pane::new(buffer))),
+			running: true,
+			buffers: buffers,
+			terminal: Terminal::new(backend).unwrap(),
 			lua: Lua::new(),
-		}
+		};
+		// a test for creating a buffer in lua and then getting the name?
+		editor.lua.globals().set("owo", Buffer::new(
+			String::from("-- This buffer is for text that is not saved, and for Lua evaluation\n-- Use this to interact with the built-in Lua interpreter.\nfunction hello()\n  print('hello, world!')\nend"),
+			String::from("*scratch*"),
+			Some(language_lua)
+		)).unwrap();
+
+		(*editor.root_pane).borrow_mut().split_window_horizontal();
+		execute!(io::stdout(), EnterAlternateScreen).unwrap();
```

Comment:
?
should this be moved into an enter() function or something more obvious? it's very hidden.

Reply:
do you mean entering the alternate screen?

Reply:
yeah, idk why it decided to show those other lines